src/muse/costs.py

```diff
@@ -112,34 +115,39 @@ def net_present_value(
 
     # Fuel/energy costs
     prices_fuel = filter_input(prices, commodity=fuels, year=years.values)
-    fuel = consumption(technologies=techs, production=production, prices=prices)
-    fuel_costs = (fuel * prices_fuel * rates).sum(("commodity", "year"))
+    fuel_costs = (cons * prices_fuel * rates).sum(("commodity", "year"))
 
     # Cost related to material other than fuel/energy and environmentals
     prices_material = filter_input(prices, commodity=material, year=years.values)
-    material_costs = (production * prices_material * rates).sum(("commodity", "year"))
+    material_costs = (cons * prices_material * rates).sum(("commodity", "year"))
 
-    # Fixed and Variable costs
-    fixed_costs = convert_timeslice(
-        techs.fix_par * (capacity**techs.fix_exp),
-        prices.timeslice,
-        QuantityType.EXTENSIVE,
-    )
-    variable_costs = techs.var_par * (
-        (production.sel(commodity=products).sum("commodity")) ** techs.var_exp
+    # Fixed costs
+    fixed_costs = (
+        convert_timeslice(
+            techs.fix_par * (capacity**techs.fix_exp),
+            prices.timeslice,
+            QuantityType.EXTENSIVE,
+        )
+        * rates
+    ).sum("year")
+
+    # Variable costs
+    tech_activity = (production / techs.fixed_outputs).max("commodity")
```

Comment:
Here we are no longer selecting the `products`, as before. I guess this is correct?

Reply:
Previously it was adding together the production of all `products`, then using this to scale the variable costs (as if the variable costs should be applied per unit of produced commodity, which is wrong). 

I'm doing something different which is calculating a ratio between commodity production and the output per unit of activity (`fixed_outputs`), and using this to scale the variable costs. In theory this ratio should be the same for all output commodities, whether `products` or `environmentals` (and nan for all commodities that aren't outputted), so when I take the max it should be fine without explicitly selecting `products`, but I guess there's no harm in doing so and it would probably be safer/more readable.